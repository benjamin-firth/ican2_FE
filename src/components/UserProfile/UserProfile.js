import React from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { fetchData } from '../../utils/apiCalls';
import { loadMessages } from '../../actions';
import './UserProfile.scss';
import PropTypes from 'prop-types';

const UserProfile = ({ user }) => {
  const currentUser = useSelector(state => state.currentUser);
  const dispatch = useDispatch();

  window.scrollTo(0, 0);

  const getMessages = () => {
    dispatch(loadMessages({}));

    const body = {"query": "{messages(sender: \""+ currentUser.id + "\", recipient: \""+ user.id + "\") {body read userId}}"}

    fetchData(body)
    .then(data => dispatch(loadMessages({
      otherMessenger: {
        id: user.id,
        name: user.name,
        pic: user.profile.image
      },
      messages: data.data.messages
    })))
    .catch(error => dispatch(loadMessages({
      otherMessenger: {
        id: user.id,
        name: user.name,
        pic: user.profile.image
      },
      messages: []
    })))
  }

  return (
    <section className='user-profile-container'>
      <section className='user-profile'>
        <div className='user-general-info'>
          <div className='user-image-container'>
            <img
              src={user.profile.image}
              alt='profile picture'
              className='profile-photo'/>
            {user.id === currentUser.id ? null : <Link to={`/messages/${user.id}`}><button id='getMessageButton' onClick={() => getMessages()}>message</button></Link>}
          </div>
          <section className='user-description-container'>
            <h2>{user.name.toLowerCase()}</h2>
            {user.mentor && <h3>{user.mentorProfile.fieldOfKnowledge}</h3>}
            {user.profile.gender !== 'none-specified' ? <p>{user.location.city}, {user.location.state} | {user.profile.gender}</p> : <p>{user.location.city}, {user.location.state}</p>}
            <p>{user.profile.aboutMe}</p>
          </section>
        </div>
        {user.mentor &&
          <section className='mentor-questions-container'>
            <h4>EXPERIENCE LEVEL</h4>
            <p>{user.mentorProfile.experienceLevel}</p>
            <h4>DESCRIBE A TYPICAL DAY AT WORK.</h4>
            <p>{user.mentorProfile.workDayQuestion}</p>
            <h4>WHAT DO YOU ENJOY MOST ABOUT YOUR WORK?</h4>
            <p>{user.mentorProfile.enjoymentQuestion}</p>
            <h4>WHAT DO YOU FEEL THE MOST EXCITED ABOUT TEACHING OTHERS?</h4>
            <p>{user.mentorProfile.teachingPointsQuestion}</p>
            <h4>WHAT IS ONE PIECE OF ADVICE YOU HAVE FOR OTHERS LOOKING TO JOIN THIS FIELD?</h4>
            <p>{user.mentorProfile.adviceQuestion}</p>
          </section>
        }
      </section>
    </section>
  )
}

UserProfile.propTypes = {
  user: PropTypes.object
}

export default UserProfile;
