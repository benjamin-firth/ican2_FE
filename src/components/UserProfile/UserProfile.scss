@import '../../variables.scss';

.user-profile-container {
  background-color: $primarycolor;
  height: 100vw;
  padding-top: 10vh;
  font-family: $primaryfont;
}

.user-profile {
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 60%;
  margin: auto;
}

.user-general-info {
  display: flex;

  img {
    border-radius: 50%;
    width: 15vw;
    height: 15vw;
  }

  h2 {
    color: $secondarycolor;
  }
}

.user-image-container {
  display: flex;
  flex-direction: column;
  justify-content: center;

  button {
    border: 5px solid $secondarycolor;
    font-family: $secondaryfont;
    font-size: 1.6vw;
    margin: 2vw .5vw 0 .5vw;
    margin-bottom: 0;
    padding: .5vw;
    width: 13vw;
    &:hover {
      background-color: $secondarycolor;
      color: white;
      cursor: pointer;
    }
  }
}

.user-description-container {
  background-color: white;
  margin: 0;
  margin-left: 1vw;
  padding: 1vw 2vw;
  display: flex;
  flex-direction: column;
  justify-content: center;

  h2, h3, p {
    margin: 0;
    text-align: center;
    margin-bottom: .5vw;
  }

  p {
    margin-bottom: 2vw;
  }

  h2 {
    font-size: 2vw;
    color: $secondarycolor;
  }
}

.mentor-questions-container {
  margin: 1vw;
  padding: 1vw 3vw;
  background-color: white;

  h4 {
    color: $secondarycolor;
    font-size: 1.3vw;
    margin-bottom: 0;
  }

  p {
    margin-top: 0;
    font-size: 1vw;
  }
}
