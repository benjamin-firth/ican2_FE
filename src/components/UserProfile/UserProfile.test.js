import React from 'react';
import { shallow } from 'enzyme';
import UserProfile from './UserProfile';

jest.mock("react-redux", () => ({
  useSelector: () => 'mockState',
  useDispatch: () => jest.fn()
}));

describe('UserProfile', () => {
  let wrapper;
  const mockUser = {
    id: 8,
    name: 'pants',
    profile: {
      image: 'url'
    },
    location: {
      city: 'Springfield',
      state: 'Ohio'
    }
  }

  beforeEach(() => {
    wrapper = shallow( <UserProfile user={mockUser}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
