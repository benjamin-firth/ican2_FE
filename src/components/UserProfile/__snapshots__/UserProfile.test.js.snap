// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match the snapshot 1`] = `
<section
  className="user-profile-container"
>
  <section
    className="user-profile"
  >
    <div
      className="user-general-info"
    >
      <div
        className="user-image-container"
      >
        <img
          alt="profile picture"
          className="profile-photo"
          src="url"
        />
        <Link
          to="/messages/8"
        >
          <button
            onClick={[Function]}
          >
            message
          </button>
        </Link>
      </div>
      <section
        className="user-description-container"
      >
        <h2>
          pants
        </h2>
        <p>
          Springfield
          , 
          Ohio
           | 
        </p>
        <p />
      </section>
    </div>
    <p>
      You are not currently listed as a mentor.
      <br />
      Want to change that? Click 
      <strong>
         edit profile 
      </strong>
       above.
    </p>
  </section>
</section>
`;
